newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js

```diff
@@ -173,26 +178,57 @@ export const HomePage = React.memo<Props>(
 
       const [activeTab, setActiveTab] = React.useState<HomeTab>(initialTab);
 
+      const { routeArguments, removeRouteArguments } = React.useContext(
+        RouterContext
+      );
+      const { setInitialPackUserFriendlySlug } = React.useContext(
+        AssetStoreContext
+      );
+      const isShopRequestedAtOpening = React.useRef<boolean>(
+        isShopRequested(routeArguments)
+      );
+
+      // Open the store and a pack or game template if asked to do so.
       React.useEffect(
         () => {
-          if (initialTab === 'get-started') {
-            incrementGetStartedSectionViewCount();
+          if (isShopRequested(routeArguments)) {
+            setActiveTab('shop');
+            if (routeArguments['asset-pack']) {
+              setInitialPackUserFriendlySlug(routeArguments['asset-pack']);
+            }
+            if (routeArguments['game-template']) {
+              setInitialGameTemplateUserFriendlySlug(
+                routeArguments['game-template']
+              );
+            }
+            // Remove the arguments so that the asset store is not opened again.
+            removeRouteArguments([
+              'initial-dialog',
+              'asset-pack',
+              'game-template',
+            ]);
           }
         },
-        [initialTab]
+        [
+          routeArguments,
+          removeRouteArguments,
+          setInitialPackUserFriendlySlug,
+          setInitialGameTemplateUserFriendlySlug,
+        ]
       );
 
       // If the user is not authenticated, the GetStarted section is displayed.
       React.useEffect(
```

Comment:
This effect scares me a bit as it really feels like tinkering.
It'll probably bit us back when we'll add new initial-tab options, but I guess we're not there yet

Reply:
I understand but it's a symptom of the design, I think that's the cleanest way to accomplish the objective of the effect